sdk/src/Services/DynamoDBv2/Custom/DataModel/_async/IDynamoDBContext.Async.cs

```diff
@@ -308,17 +387,54 @@ partial interface IDynamoDBContext
+        [Obsolete("Use the QueryAsync overload that takes QueryConfig instead, since DynamoDBOperationConfig contains properties that are not applicable to QueryAsync.")]
         AsyncSearch<T> QueryAsync<T>(object hashKeyValue, QueryOperator op, IEnumerable<object> values, DynamoDBOperationConfig operationConfig = null);
 
+        /// <summary>
+        /// Configures an async Query operation against DynamoDB, finding items
+        /// that match the specified range element condition for a hash-and-range primary key.
+        /// </summary>
+        /// <typeparam name="T">Type of object.</typeparam>
+        /// <param name="hashKeyValue">Hash key of the items to query.</param>
+        /// <param name="op">Operation of the condition.</param>
+        /// <param name="values">
+        /// Value(s) of the condition.
+        /// For all operations except QueryOperator.Between, values should be one value.
+        /// For QueryOperator.Betwee, values should be two values.
+        /// </param>
+        /// <param name="queryConfig">Config object that can be used to override properties on the table's context for this request.</param>
+        /// <returns>AsyncSearch which can be used to retrieve DynamoDB data.</returns>
+        AsyncSearch<T> QueryAsync<T>(object hashKeyValue, QueryOperator op, IEnumerable<object> values, QueryConfig queryConfig);
+
+        /// <summary>
+        /// Configures an async Query operation against DynamoDB using a mid-level document model 
+        /// query configration, finding items that match the specified conditions. 
+        /// </summary>
+        /// <typeparam name="T">Type of object.</typeparam>
+        /// <param name="queryConfig">Mid-level, document model query request object.</param>
+        /// <returns>AsyncSearch which can be used to retrieve DynamoDB data.</returns>
+        AsyncSearch<T> FromQueryAsync<T>(QueryOperationConfig queryConfig);
+
         /// <summary>
         /// Configures an async Query operation against DynamoDB, finding items
         /// that match the specified conditions.
         /// </summary>
         /// <typeparam name="T">Type of object.</typeparam>
-        /// <param name="queryConfig">Query request object.</param>
+        /// <param name="queryOperationConfig">Mid-level, document model query request object.</param>
         /// <param name="operationConfig">Config object which can be used to override the table used.</param>
         /// <returns>AsyncSearch which can be used to retrieve DynamoDB data.</returns>
-        AsyncSearch<T> FromQueryAsync<T>(QueryOperationConfig queryConfig, DynamoDBOperationConfig operationConfig = null);
+        [Obsolete("Use the FromQueryAsync overload that takes QueryConfig instead, since DynamoDBOperationConfig contains properties that are not applicable to FromQueryAsync.")]
+        AsyncSearch<T> FromQueryAsync<T>(QueryOperationConfig queryOperationConfig, DynamoDBOperationConfig operationConfig = null);
```

Comment:
Renaming the argument from `queryConfig` -> `queryOperationConfig` would be a breaking change

Reply:
Reset in https://github.com/aws/aws-sdk-net/pull/3383/commits/987fe702d9428657cff4228bf91457a1ae9dcd74, and names are now less ambiguous after the adjustments due to this thread: https://github.com/aws/aws-sdk-net/pull/3383#discussion_r1673120405